src/Slices/Home/UI/Page.tsx

```diff
@@ -1,50 +1,76 @@
-import React, { useContext } from "react";
+import React from "react";
 import { PageSection } from "@patternfly/react-core";
-import { RemoteData } from "@/Core";
-import { DependencyContext, words } from "@/UI";
+import { Environment } from "@/Core";
+import { useGetEnvironments } from "@/Data/Managers/V2/Environment/GetEnvironments/useGetEnvironments";
+import { useGetProjects } from "@/Data/Managers/V2/Project/GetProjects/useGetProjects";
+import { words } from "@/UI";
 import { ErrorView, LoadingView, PageSectionWithTitle } from "@/UI/Components";
 import { EnvironmentsOverview } from "./EnvironmentsOverview";
 import { EmptyFilterToolbar } from "./FilterToolbar";
 
+/**
+ * Home page
+ *
+ * @returns {React.FC} The Home page
+ */
 export const Page: React.FC = () => {
-  const { queryResolver } = useContext(DependencyContext);
-  const [data] = queryResolver.useContinuous<"GetEnvironmentsContinuous">({
-    kind: "GetEnvironmentsContinuous",
-    details: true,
-  });
+  const { data, isSuccess, isError, error, refetch } = useGetEnvironments().useContinuous();
+  const projects = useGetProjects().useContinuous();
+
+  if (isError) {
+    return (
+      <>
+        <PageSectionWithTitle title={words("home.title")} />
+        <EmptyFilterToolbar />
+        <PageSection hasBodyWrapper={false}>
+          <ErrorView
+            title={words("error")}
+            message={words("error.general")(error.message)}
+            ariaLabel="Overview-Failed"
+            retry={refetch}
+          />
+        </PageSection>
+      </>
+    );
+  }
+  if (projects.isError) {
+    return (
+      <>
+        <PageSectionWithTitle title={words("home.title")} />
+        <EmptyFilterToolbar />
+        <PageSection hasBodyWrapper={false}>
+          <ErrorView
+            title={words("error")}
+            message={words("error.general")(projects.error.message)}
+            ariaLabel="Overview-Failed"
+            retry={refetch}
+          />
+        </PageSection>
+      </>
+    );
+  }
+
+  if (isSuccess && projects.isSuccess) {
+    const envsWithProjectName: Environment[] = data.map((env) => ({
```

Comment:
```TS
const projectMap = new Map(projects.data.map((project) => [project.id, project.name]));
const envsWithProjectName: Environment[] = data.map((env) => ({
  ...env,
  projectName: projectMap.get(env.project_id) || "",
}));
```
This avoids repeatedly iterating over the `projects.data` array for each environment.